tests/test_pytypes.cpp

```diff
@@ -122,14 +122,14 @@ void m_defs(py::module_ &m) {
 namespace literals {
 enum Color { RED = 0, BLUE = 1 };
 
-typedef py::typing::Literal<"26",
-                            "0x1A",
-                            "\"hello world\"",
-                            "b\"hello world\"",
-                            "u\"hello world\"",
-                            "True",
-                            "Color.RED",
-                            "None">
+typedef py::typing::Literal<26,
+                            0x14,
+                            py::typing::StringLiteral("\"hello world\""),
+                            py::typing::StringLiteral("b\"hello world\""),
+                            py::typing::StringLiteral("u\"hello world\""),
+                            true,
+                            py::typing::StringLiteral("Color.RED"),
+                            py::typing::StringLiteral("None")>
```

Comment:
Just skimming over this, but would a simple string literal than have to use:
`py::typing::Literal<py::typing::StringLiteral("hello")>`?
Would be nice if strings could be used directly.

Reply:
What could be done I believe is only allow each Literal of the same type. Then I don't think you would need the `py::typing::Literal<py::typing::StringLiteral("hello")>` however, you wouldn't be able to mix non-type templates inside Literals and would need to use Unions.

Reply:
After messing around with various solutions the only one I could get working is make different named literals.

```C++
py::typing::LiteralString"Hi">
py::typing::LiteralInt<1, 2, 3>
py::typing::LiteralBool<true>
```

To avoid confusion if we wanted to move forward with this the current `typing::StringLiteral` would be change to `typing::fixed_string`,